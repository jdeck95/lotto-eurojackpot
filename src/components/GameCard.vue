<script setup lang="ts">
interface Props {
  title: string;
  route: string;
  buttonLabel: string;
}

const props = defineProps<Props>();
</script>

<template>
  <div class="column eurojackpot text-h3 col-11 col-sm-6 col-md-5 col-lg-3 q-px-lg rounded-borders">
    <div class="row justify-center q-mt-xl">{{ props.title }}</div>
    <q-btn
      :label="props.buttonLabel"
      outline
      color="white"
      class="q-my-xl"
      :to="props.route"
    />
  </div>
</template>
