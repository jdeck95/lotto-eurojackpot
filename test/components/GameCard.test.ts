import { mount, type VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { Quasar } from 'quasar';
import { beforeEach, describe, expect, it } from 'vitest';
import GameCard from '../../src/components/GameCard.vue';

describe('GameCard', () => {
  let wrapper: VueWrapper<any>;

  const pinia = createTestingPinia();
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(GameCard, {
      global: {
        plugins: [Quasar, pinia],
      },
      props: {
        title: 'Game Title',
        route: '/game',
        buttonLabel: 'button',
      },
    });
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
