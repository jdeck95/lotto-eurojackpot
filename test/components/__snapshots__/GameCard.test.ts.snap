// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameCard > should render correctly 1`] = `
"<div class="column eurojackpot text-h3 col-11 col-sm-6 col-md-5 col-lg-3 q-px-lg rounded-borders">
  <div class="row justify-center q-mt-xl">Game Title</div><button class="q-btn q-btn-item non-selectable no-outline q-btn--outline q-btn--rectangle text-white q-btn--actionable q-focusable q-hoverable q-my-xl" tabindex="0" type="button"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row"><span class="block">button</span></span></button>
</div>"
`;
